plc4j/spi/src/main/java/org/apache/plc4x/java/spi/Plc4xNettyWrapper.java

```diff
@@ -231,19 +236,21 @@ public Duration getTimeout() {
             completionCallback.andThen(handler.getPacketConsumer()),
             handler.getOnTimeoutConsumer(),
             handler.getErrorConsumer(),
+            handler::confirmHandled,
+            handler::confirmError,
+            handler::cancel,
             handler.getTimeout()
         );
         deferred.set(registration);
         registeredHandlers.add(registration);
     }
 
     private Consumer<TimeoutException> onTimeout(AtomicReference<HandlerRegistration> reference, Consumer<TimeoutException> onTimeoutConsumer) {
-        return new Consumer<TimeoutException>() {
-            @Override
-            public void accept(TimeoutException e) {
-                registeredHandlers.remove(reference.get());
-                onTimeoutConsumer.accept(e);
-            }
+        return timeoutException -> {
+            final HandlerRegistration registration = reference.get();
```

Comment:
Just a note, syntax sugar makes stack traces harder to follow. Explicit creation of `new Consumer` was intentional to keep stacktrace more obvious.

Reply:
OK, I wasn't aware of that. I'll change it back, and leave a comment.